flow/layers/physical_shape_layer.cc

```diff
@@ -51,11 +54,29 @@ void PhysicalShapeLayer::Preroll(PrerollContext* context,
     set_needs_system_composite(true);
 #else
     // Add some margin to the paint bounds to leave space for the shadow.
-    // The margin is hardcoded to an arbitrary maximum for now because Skia
-    // doesn't provide a way to calculate it.  We fill this whole region
-    // and clip children to it so we don't need to join the child paint bounds.
+    // We fill this whole region and clip children to it so we don't need to
+    // join the child paint bounds.
+    // The offset is calculated as follows:
+
+    //                   .--                            (kLightRadius = 800)
+    //                 -----                            (light)
+    //                   |                              (kLightHeight = 600)
+    //             -------------                        (layer)
+    //                   |
+    //                   |                              (elevation)
+    //                   |
+    // ------------------------------------------------ (canvas)
+    //  -----------                                     (extent of shadow)
+    //
+    // E = lx        }           x = (r + w/2)/h
+    //                } =>
+    // r + w/2 = hx  }           E = (l/h)(r + w/2)
```

Comment:
Looks good! It would be better to add a few legends about what `E, x, h, l` mean.

Reply:
Done